src/inet_address.rs

```diff
@@ -38,3 +40,49 @@ impl InetAddress {
         }
     }
 }
+
+pub trait InetAddressExtManual {
+    fn to_bytes<'a>(&'a self) -> Option<InetAddressBytes<'a>>;
+}
+
+impl<O: IsA<InetAddress>> InetAddressExtManual for O {
+    /// Returns `None` in case the address has a native size different than 4 and 16.
+    fn to_bytes<'a>(&'a self) -> Option<InetAddressBytes<'a>> {
+        let size = self.get_native_size();
+        unsafe {
+            let bytes = gio_sys::g_inet_address_to_bytes(self.as_ref().to_glib_none().0);
+            if size == 4 {
+                Some(InetAddressBytes::V4(&*(bytes as *const [u8; 4])))
+            } else if size == 16 {
+                Some(InetAddressBytes::V6(&*(bytes as *const [u8; 16])))
+            } else {
+                None
+            }
+        }
+    }
+}
+
+impl From<IpAddr> for InetAddress {
+    fn from(addr: IpAddr) -> Self {
+        match addr {
+            IpAddr::V4(v4) => InetAddress::from_bytes(InetAddressBytes::V4(&v4.octets())),
+            IpAddr::V6(v6) => InetAddress::from_bytes(InetAddressBytes::V6(&v6.octets())),
+        }
+    }
+}
+
+impl Into<IpAddr> for InetAddress {
```

Comment:
Shouldn't this rather be a fallible conversion, instead of panicking? Also, is relying on the returned size like that safe when from the looks of it the `InetAddress` type can also represent unix paths that don't have a fixed length?

Reply:
This can't really fail though, `InetAddress` / `IpAddr` is either an IPv4 or IPv6 address and nothing else.

UNIX addresses are `GUnixSocketAddress`

Reply:
Actually nevermind the second part, I got confused by the documentation.